### PR TITLE
[the Lens] New translator

### PR DESCRIPTION
**Pull request by eurruby**
This is a beta translator for https://www.lens.org/

### CODE REVIEW

File: the Lens.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	newItem.filingDate = filingDate.toISOString().substr(0,10);
+
+	var ul = doc.getElementById("inventorFilter");
+	var liNodes = [];
+	for (var i = 0; i < ul.childNodes.length; i++) {
+		if (ul.childNodes[i].nodeName == "LI") {
+			liNodes.push(ul.childNodes[i]);
+		}
+	}
+	iAu = liNodes.length;
```
Comment (zuphilip) [inline the Lens.js]: It seems for me that you don't need much of these...

I suggest to do something like
```js
var inventors = ZU.xpath(doc, '//*[@id="inventorFilter"]/li/a');
for (let inventor of inventors) {
   newItem.creators.push(ZU.cleanAuthor(inventor, "inventor", false);
}
```
(not tested)

File: the Lens.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if(url.indexOf("search?") !== -1) {
+		return "multiple";
+	}
+		
+	if (url.indexOf("patent") !== -1 && getTitle(doc)) {
+		return "patent";
```
Comment (zuphilip) [inline the Lens.js]: What about some of the other subpages like https://www.lens.org/lens/patent/109-941-370-099-325/citations  ?

File: the Lens.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+	newItem.creators.push(Zotero.Utilities.cleanAuthor(authors[iAu], "inventor", true));
+	}
+
+	var pdfurl = ZU.xpathText(doc, '//*[@id="patentFrontPage"]/div[2]/a/@href');
+	newItem.attachments.push({
+		title: 'Full Text PDF',
+		url: pdfurl,
+		mimeType: 'application/pdf'
+	});
```
Comment (zuphilip) [inline the Lens.js]: We should test here that `pdfurl` has some contents and only then do these lines here.

**Comment by eurruby** [comment]
no sorry I can't because I changed job

**Comment by AbeJellinek** [comment]
The Lens now has COinS metadata that Zotero scrapes quite well. Closing this.

**Pull request closed without merging**